### ISSUE TITLE
OOM RuntimeError is raised due to the huge memory cost during label assignment.CPU mode is applied in this batch. 

### ISSUE THREAD

**Issue by np-n** [meituan/YOLOv6#656]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

I want to train the table cell detector using yolov6. My dataset has a large number of labels for a single image. But throws the error `huge memory cost during label assignment`. I have tried to reduce the batch size to 16/8, but It also stuck during the first iterations. Do you have any suggestions to fix this issue?
 
`tcmalloc: large alloc 4512153600 bytes == 0x7f3159308000 @  0x7f3d43ccfb6b 0x7f3d43cef379 0x7f3cc9e59935 0x7f3cc9e37093 0x7f3cf6f0453f 0x7f3cf6f05360 0x7f3cf6f053b4 0x7f3cf75a640f 0x7f3cf7ff564b 0x7f3cf7cb5f47 0x7f3cf7fc826f 0x7f3cf7cf6e22 0x7f3cf75acb76 0x7f3cf8182065 0x7f3cf79ec9a2 0x7f3cf7fc8505 0x7f3cf7a49901 0x7f3cf7439fc7 0x7f3cf833fe00 0x7f3cf7a26964 0x7f3d1faaded0 0x5d80be 0x5d8d8c 0x561f80 0x4fd2db 0x4997c7 0x5d8868 0x5d8506 0x55f797 0x55d078 0x5d8941
     0/199     1.793         0     1.005:  28% 100/361 [03:36<17:00,  3.91s/it]OOM RuntimeError is raised due to the huge memory cost during label assignment.                     CPU mode is applied in this batch. If you want to avoid this issue,                     try to reduce the batch size or image size.`

**Comment by jaideep11061982** [comment]
i too have this issue @cfc4n @yeldarby @rainsun @zhiyelee

**Comment by wenlihaoyu** [comment]
> * apply

跟踪代码，应该是labels太多，超过了显卡显存，当前batch就会走cpu 训练。https://github.com/meituan/YOLOv6/blob/main/yolov6/models/loss.py#L99

and

**Comment by np-n** [comment]
@wenlihaoyu, I have reduced the batch size to 8 and the image size to 416*416. But it throws the same error after 50 epochs. Is there any other solution to fix it, without increasing the resources?

**Comment by wenlihaoyu** [comment]
> @wenlihaoyu, I have reduced the batch size to 8 and the image size to 416*416. But it throws the same error after 50 epochs. Is there any other solution to fix it, without increasing the resources?
主要这一步会造成显卡溢出，https://github.com/meituan/YOLOv6/blob/99a6f5824672606b18c58638c7ab6a4e802a5b4f/yolov6/assigners/assigner_utils.py#L42
@np-n s 如果单卡训练,   修改训练策略，更小的batch_size,  梯度累加。

**Comment by np-n** [comment]
@wenlihaoyu, how to do gradient accumulation?

**Comment by Chilicyy** [comment]
@np-n When batchsize is less than 64, gradient accumulation will take effect. You can change the value by yourself as well. 
https://github.com/meituan/YOLOv6/blob/99a6f5824672606b18c58638c7ab6a4e802a5b4f/yolov6/core/engine.py#L301
